src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -560,16 +560,17 @@ private void setupTimer(long initialRecurrencePeriod) {
 
         @Override public void output(InputStream stream) throws Exception {
             PrintStream ps = listener.getLogger();
-            OutputStream os;
-            if (ps.getClass() == PrintStream.class) {
-                // Try to extract the underlying stream, since swallowing exceptions is undesirable and PrintStream.checkError is useless.
-                os = (OutputStream) printStreamDelegate.get(ps);
-            } else {
-                // A subclass. Who knows why, but trust any write(…) overrides it may have.
-                os = ps;
-            }
             try {
-                IOUtils.copy(stream, os);
+                if (ps.getClass() == PrintStream.class) {
+                    // Try to extract the underlying stream, since swallowing exceptions is undesirable and PrintStream.checkError is useless.
+                    OutputStream os = (OutputStream) printStreamDelegate.get(ps);
+                    synchronized (os) { // like PrintStream.write overloads do
```

Comment:
Did you mean `synchronized (ps)`?

Reply:
Indeed that seems to be the lock, will change.